src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -588,12 +588,12 @@ public static int getNumberOfChunks(long size) {
         long size = getSize();
         long newSize = LongUtil.intsToLong(endHigh, endLow);
         return network.synchronizer.applyComplexUpdate(owner(), signingPair(),
-                (s, committer) -> overwriteSection(s, committer, fileData,
-                        0L, newSize, network, crypto, monitor)
+                (s, committer) -> clean(s, committer, network, crypto)
+                    .thenCompose(u -> u.left.overwriteSection(u.right, committer, fileData,
+                            0L, newSize, network, crypto, monitor)
```

Comment:
Still not closing the scope early. line 593 should end the clean thencompose

Reply:
fixed